# Recent Holiday Trends and Traveller Behaviour Survey


## The purpose of this survey is to explore recent holiday trends and traveller behaviour. This survey forms part of an academic project for the Statistical Techniques for Data Analytics module.

In [77]:
import pandas as pd

## Load the Dataset

In [78]:
df_Holidays = pd.read_csv("Holiday_trends_Data_CA1.csv")

### Display first 5 rows, to ensure the data has correctly loaded


In [79]:
df_Holidays.head(5)

,Timestamp,\nWhat age range are you in?,Was your holiday,What was the main purpose of your Holiday,How did you book your holiday,How recent was this holiday,How would you describe your holiday type,How many people were in your travel group?,Approximately how much did your accommodation cost per room per night (€)?,How many nights did your holiday last?,"Overall Holiday Cost - Approximate total spend for all travellers (€), including cost of holiday + daily spending.",How satisfied were you with your holiday overall?,Would You Take This Type of Holiday Again?,Did You Travel by Air?,How many holidays have you taken in the last 12 months?
0,25/09/2026 12:25,65+,A Holiday abroad (outside Ireland),"To attend an event (concert, Football match etc)",Independently Booked online,Greater than 3 months ago but within the last ...,A holiday with friends,6,100,3,800,5,Yes,Yes,2
1,25/09/2026 12:25,55-64,A Holiday abroad (outside Ireland),Beach Holiday abroad,A package holiday,In last 3 months,A family holiday,2,150,5,3000,3,Yes,Yes,3
2,25/09/2026 12:27,65+,A Holiday abroad (outside Ireland),Beach Holiday abroad,Independently Booked online,In last 3 months,A family holiday,4,70,28,4800,5,Yes,No,4
3,25/09/2026 12:29,65+,A Holiday abroad (outside Ireland),City Break,Independently Booked online,Greater than 3 months ago but within the last ...,A holiday with friends,6,50,3,500,5,Yes,Yes,2
4,25/09/2026 12:32,25-34,A Holiday abroad (outside Ireland),Other,Independently Booked online,Greater than 3 months ago but within the last ...,A holiday with friends,3,115,6,3000,5,Yes,Yes,3


###  Trim any spaces from the headings

In [80]:
df_Holidays.columns = df_Holidays.columns.str.strip()

# Rename columns to make it more readable 

In [81]:
df_Holidays = df_Holidays.rename(columns={
    "Timestamp": "Timestamp",
    "What age range are you in": "Age_Range",
    "Was your holiday": "Holiday_Location",
    "What was the main purpose of your Holiday": "Holiday_Purpose",
    "How did you book your holiday": "Booking_Method",
    "How recent was this holiday": "Holiday_Recency",
    "How would you describe your holiday type": "Holiday_Type",
    "How many people were in your travel group?": "Group_Size",
    "Approximately how much did your accommodation cost per room per night (€)?": "Accommodation_Cost",
    "How many nights did your holiday last?": "Holiday_Length",
    "Overall Holiday Cost - Approximate total spend for all travellers (€), including  cost of holiday + daily spending.": "Total_Cost",
    "How satisfied were you with your holiday overall?": "Satisfaction",
    "Would You Take This Type of Holiday Again?": "Holiday_Again",
    "Did You Travel by Air?": "Travel_By_Air",
    "How many holidays have you taken in the last 12 months?": "Holidays_Last_12_Months"
    } 
    )

## Show Statistics on the csv file

###  Descriptive Statistics

In [82]:
df_Holidays.describe().T

,count,mean,std,min,25%,50%,75%,max
Group_Size,48.0,3.541667,1.923631,1.0,2.0,3.0,4.00,10.0
Accommodation_Cost,48.0,221.104167,575.275114,10.0,77.5,112.5,150.00,4000.0
Holiday_Length,48.0,7.562500,5.675278,2.0,4.0,7.0,7.25,28.0
Total_Cost,48.0,2427.083333,1683.776099,200.0,950.0,2000.0,3125.00,6000.0
Satisfaction,48.0,4.416667,0.678965,3.0,4.0,5.0,5.00,5.0
Holidays_Last_12_Months,48.0,2.520833,1.129669,1.0,2.0,2.0,3.00,5.0


In [83]:
df_Holidays.shape

(48, 15)

### Check for missing values

In [84]:
df_Holidays.isnull().sum()

Timestamp                     0
What age range are you in?    0
Holiday_Location              0
Holiday_Purpose               0
Booking_Method                0
Holiday_Recency               0
Holiday_Type                  0
Group_Size                    0
Accommodation_Cost            0
Holiday_Length                0
Total_Cost                    0
Satisfaction                  0
Holiday_Again                 0
Travel_By_Air                 0
Holidays_Last_12_Months       0
dtype: int64

### No missing values were identified in the dataset. Therefore no records required removal

### Show Skewness of the data

In [85]:
df_Holidays.skew(numeric_only=True)

Group_Size                 1.421523
Accommodation_Cost         6.337030
Holiday_Length             2.474319
Total_Cost                 0.621934
Satisfaction              -0.748730
Holidays_Last_12_Months    0.500526
dtype: float64

### Accomodation Costs shows a positive skewness (6.33), indicating the presence of extreme observations. In Descriptive Statistics, the average accomodation cost is €221.10, the maximum recorded value is €4,000, suggesting a small number of very expensive holidays are influencing the overall distribution

### Holiday Length also demonstrates a positive skewness of 2.47.  While most holidays show an average of 7.5 days, the maximum holiday length of 28 days was recorded, again this outlier is influencing the overall distribution

### The Satisfaction skewness of (-0.74) indicates that responses are clustered towards the higher end of the rating scale.  This is suggesting that people were generally satisfied with their holiday experience

In [86]:
df_Holidays.median(numeric_only=True)

Group_Size                    3.0
Accommodation_Cost          112.5
Holiday_Length                7.0
Total_Cost                 2000.0
Satisfaction                  5.0
Holidays_Last_12_Months       2.0
dtype: float64

### See above median values.  Accomodation costs has a median of €112.50, however the mean is €221.10.  This further demonstrates the impact of extreme values on the distribution